jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -285,18 +285,23 @@ def _on_flux_unit_changed(self, msg):
         ):
             self.pixar_sr_exists = False
 
-    def _find_and_convert_contour_units(self, yunit=None):
+    def _find_and_convert_contour_units(self, msg=None, yunit=None):
         if not yunit:
             yunit = self.sb_unit_selected
 
+        if msg is not None:
+            viewers = [self.app.get_viewer(msg.viewer_reference)]
+        else:
+            viewers = [viewer._obj for _, viewer in self._app._jdaviz_helper.viewers.items()]
```

Comment:
```suggestion
            viewers = self._app._viewer_store.values()
```

This will give dict values back, but since you're just iterating over it, it probably doesn't need to be casted to a list 🤷‍♂️